fairlearn/metrics/_exposure.py

```diff
@@ -0,0 +1,304 @@
+        y_pred,
+        *,
+        sensitive_features,
+        method='between_groups',
+        sample_weight=None) -> float:
+    """Calculate the difference in exposure allocation between groups.
+
+    An exposure allocation difference is defined as the difference
+    between the largest and the smallest group-level exposure.
+    The exposure allocation difference of 0 means that all groups have the same exposure. A high
+    exposure difference can be seen as an indication of allocation harm.
+
+
+    Parameters
+    ----------
+    y_true : array-like
+        Ground truth relevance scores.
+    y_pred : array_like of unique integers
+        The predicted ranking. The lower the integer the higher on the ranking.
+    sensitive_features :
+        The sensitive features over which the allocation harm should be assessed
+    method : str
+        How to compute the differences. See :func:`fairlearn.metrics.MetricFrame.difference`
+        for details.
+    sample_weight : array-like
+        The sample weights
+
+    Returns
+    -------
+    float
+        The exposure difference
+    """
+    exposure_mf = MetricFrame(metrics=exposure,
+                              y_true=y_true,
+                              y_pred=y_pred,
+                              sensitive_features=sensitive_features,
+                              sample_params={'sample_weight': sample_weight})
+    result = exposure_mf.difference(method=method)
+    return result
+
+
+def exposure_ratio(
+        y_true,
+        y_pred,
+        *,
+        sensitive_features,
+        method='between_groups',
+        sample_weight=None) -> float:
+    """Calculate the ratio of exposure allocation between groups.
+
+    The exposure ratio is defined as the ratio
+    between the largest and the smallest group-level exposure.
+    The exposure allocation ratio of 1 means that all groups have the same exposure. A low exposure
+    ratio can be seen as an indication of allocation harm.
+
+
+    Parameters
+    ----------
+    y_true : array-like
+        Ground truth relevance scores.
+    y_pred : array_like of unique integers
+        The predicted ranking. The lower the integer the higher on the ranking.
+    sensitive_features :
+        The sensitive features over which the allocation harm should be assessed
+    method : str
+        How to compute the differences. See :func:`fairlearn.metrics.MetricFrame.ratio`
+        for details.
+    sample_weight : array-like
+        The sample weights
+
+    Returns
+    -------
+    float
+        The exposure ratio
+    """
+    exposure_mf = MetricFrame(metrics=exposure,
+                              y_true=y_true,
+                              y_pred=y_pred,
+                              sensitive_features=sensitive_features,
+                              sample_params={'sample_weight': sample_weight})
+    result = exposure_mf.ratio(method=method)
+    return result
+
+
+def proportional_exposure_difference(
```

Comment:
This one should also follow the API for the ranking metrics.

I think that we may want to consider adjusting the definition as they did in Eq. (4) here:
* [Policy Learning for Fairness in Ranking](https://papers.nips.cc/paper/2019/file/9e82757e9a1c12cb710ad680db11f6f1-Paper.pdf)